### PR TITLE
Arcade that sends you to OpenDream Paradise station

### PR DESCRIPTION
**Pull request by Simyon264**
## About the PR

Adds an arcade that, when used, sends you to paradise station. It forces you to sleep for 60 seconds while adding a little text upon examination that indicates that you are in paradise station.
## Why / Balance

Suggestion from #contributors. 
I do have indirect permission from ike709, moderator and contributor for open dream, to connect people to open dream servers. But I have no idea if that is enough. I asked around on the Open Dream discord server and will update, till then I will keep this PR as a draft.

See [this](https://canary.discord.com/channels/310555209753690112/770682801607278632/1222387386034556969) discord message.

Update: According to amylizzle, the open dream servers are hosted by PJB, which means this is allowed per Hub rules. Hooray! Also because amylizzle said that OpenDream is administered largely by wixoa, I also asked him.

Update 2: Got permissions from wixoa.
image

## Technical details

Honestly, this is the first time I ever coded UI, and I am so sure that I produced shitcode. But I hope it's good enough for an april fools joke.

## Media

https://github.com/space-wizards/space-station-14/assets/63975668/90fc36b4-40f8-4d45-8096-bb2878e1b7f2

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl: Simyon
- add: There are new arcade machines around that let you play space station 13!

### CODE REVIEW

**Comment by Simyon264** [comment]
This adds it to the random arcade spawner, meaning it will spawn in with the other arcades. At least on most maps.

**Comment by Flareguy** [comment]
> This adds it to the random arcade spawner, meaning it will spawn in with the other arcades. At least on most maps.

I immediately noticed my mistake after posting, oops.
Anyways, you could probably still serve to weigh this more then the others by putting them in a rare prototype group. You'd probably want more people to play it anyways since it's only for april fools.

**Comment by Simyon264** [comment]
> Anyways, you could probably still serve to weigh this more then the others by putting them in a rare prototype group. You'd probably want more people to play it anyways since it's only for april fools.

How? I am unsure how they work.

**Comment by Flareguy** [comment]
Add

> > Anyways, you could probably still serve to weigh this more then the others by putting them in a rare prototype group. You'd probably want more people to play it anyways since it's only for april fools.
> 
> How? I am unsure how they work.

To my understanding, from looking at the maint loot spawner: you need to add a ``rarePrototypes`` field and set ``rareChance`` under it. You should probably set ``chance`` to make up the difference of whatever you have rarePrototypes set to, otherwise I think there's a chance the spawner will just spawn nothing.
You'd be looking at something like this, to my understanding:
```  - type: RandomSpawner
    prototypes:
      - ParadiseArcade
      chance: 0.50
    rarePrototypes:
      - BlockGameArcade
      - SpaceVillainArcadeFilled
      rareChance: 0.50

**Comment by Simyon264** [comment]
Alright, thanks. Will implement soon.

File: Resources/Textures/Structures/Machines/arcade.rsi/paradise.png
Comment (VasilisThePikachu) [inline Resources/Textures/Structures/Machines/arcade.rsi/paradise.png:1]: This is a little over 32x32

**Review by VasilisThePikachu — CHANGES_REQUESTED** [review]

File: Resources/Textures/Structures/Machines/arcade.rsi/paradise.png
Comment (Simyon264) [reply Resources/Textures/Structures/Machines/arcade.rsi/paradise.png:1]: How did I manage that lmao

File: Resources/Locale/en-US/arcade/paradise.ftl [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿paradise-arcade-on-examine = [color=lightblue]{CAPITALIZE(SUBJECT($target))} {CONJUGATE-BE($target)} dreaming about paradise...[/color]
+paradise-arcade-window-title = Paradise Station
+paradise-arcade-button-label = Are you sure? This will connect you to another server.
```
Comment (VasilisThePikachu) [inline Resources/Locale/en-US/arcade/paradise.ftl:3]: I believe this message may need to be more verbose so that this will close your current client.

**Comment by VasilisThePikachu** [comment]
> the open dream servers are hosted by PJB, which means this is allowed per Hub rules

Still wanna see permission from her and a quick glance that this is fine.

@PJB3005 require your review.

File: Resources/Textures/Structures/Machines/arcade.rsi/paradise.png
Comment (Simyon264) [reply Resources/Textures/Structures/Machines/arcade.rsi/paradise.png:1]: Changing the image in any way is breaking shit. I don't know how to change it.

File: Resources/Locale/en-US/arcade/paradise.ftl (hunk 1 above)
Comment (Simyon264) [reply Resources/Locale/en-US/arcade/paradise.ftl]: Done.

**Comment by K-Dynamic** [comment]
Imagine being able to cross-message between servers

**Comment by Simyon264** [comment]
Permission from PJB aquired.

File: Content.Client/Arcade/UI/ParadiseArcadeBoundUserInterface.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using System.Threading.Tasks;
+using Content.Shared.Arcade;
+using JetBrains.Annotations;
+using Robust.Client;
+using Robust.Shared.Timing;
+using Serilog;
```
Comment (metalgearsloth) [inline Content.Client/Arcade/UI/ParadiseArcadeBoundUserInterface.cs:6]: Wrong logger but not master so don't care.

Comment (Simyon264) [reply Content.Client/Arcade/UI/ParadiseArcadeBoundUserInterface.cs:6]: Oh yeah. That was very much just temporary. Then it became permanent

**Pull request merged**